lib/routes/dhu/xxgk/news.ts

```diff
@@ -25,25 +26,54 @@ export const route: Route = {
 };
 
 async function handler() {
-    const link = base_url;
-    const response = await got({
-        method: 'get',
-        url: link,
-        headers: {
-            Referer: base_url,
-        },
-    });
-
-    const $ = load(response.data);
+    const link = baseUrl;
+    const { data: response } = await got(link);
+
+    const $ = load(response);
+
+    // article list
+    const articleList = $('.cols_list > li')
+        .toArray()
+        .map((item) => {
+            item = $(item);
+            const cols_title = item.find('.cols_title > a');
+            const cols_meta = item.find('.cols_meta');
+
+            const link = cols_title.attr('href');
+            const title = cols_title.text();
+            const pubDate = parseDate(cols_meta.text(), 'YYYY-MM-DD', 'zh-cn');
+            return {
+                title,
+                link,
+                pubDate,
+                description: '',
+            };
+        });
+
+    // fetch article content
+    const items = await Promise.all(
+        articleList.map(async (item) => {
+            const url = `${siteUrl}${item.link}`;
```

Comment:
You can do this in
https://github.com/DIYgod/RSSHub/blob/1460a8447d840970d7791278301f1fe6738f5ddd/lib/routes/dhu/xxgk/news.ts#L45-L50